### PR TITLE
Update of the sld calculator documentation. Fixes #2689.

### LINKED ISSUE #2689: Improve documentation for SLD calculator
**Is your feature request related to a problem? Please describe.**
The documentation for the SLD calculator is a bit thin and could benefit with more use cases that are allowed by the `periodictable` package.

**Describe the solution you'd like**
The [NIST Activation and Scattering Calculator](https://www.ncnr.nist.gov/resources/activation/) has extensive documentation that would be easy to integrate.

**Additional context**
A few useful cases not covered in our documentation but covered in the online calculator:

- Mass and isotopic densities in the chemical formulas (**_H2O@1_** -or- **_D2O@1n_**)
- Mole and mass fractions (**_78.2H2O[16] + 21.8H2O[18]_** -or- **_50%wt Co // Ti_**)
- Biomacromolecules ("code:sequence" where code can be 'aa', 'dna', or 'rna' e.g. **_aa:RELEELNVPGEIVESLSSSEESITRINKKIEKFQSEEQQQTEDELQDKIHPFAQTQSLVYPFPGPIPNSLPQNIPPLTQTPVVVPPFLQPEVMGVSKVKEAMAPKHKEMPFPKYPVEPFTESQSLTLTDVENLHLPLPLLQSWMHQPHQPLPPTVMFPPQSVLSLSQSKVLPVPQKAVPYPQRDMPIQAFLLYQEPVLGPVRGPFPIIV_**)

This came from a discussion with @katieweigandt

### PR DESCRIPTION
**Pull request by katieweigandt**
## Description

Update of the SLD calculator tool to describe different ways of entering in material information. 

Fixes # 2689

## How Has This Been Tested?

Viewed RST through pycharm.

## Review Checklist (please remove items if they don't apply):

- [ ] Code has been reviewed
- [ ] Functionality has been tested
- [ ] Windows installer (GH artifact) has been tested (installed and worked) 
- [ ] MacOSX installer (GH artifact) has been tested (installed and worked) 
- [ ] User documentation is available and complete (if required)
- [ ] Developers documentation is available and complete (if required)
- [ ] The introduced changes comply with SasView license (BSD 3-Clause)

### CODE REVIEW

File: src/sas/qtgui/Calculators/media/sld_calculator_help.rst
Comment (krzywon) [inline src/sas/qtgui/Calculators/media/sld_calculator_help.rst:22]: Potential wordsmithing:
The section "Specifying Materials or Mixtures in the Molecular Formula Field" offers further guidance on how to format molecules, biomolecules, and more complex mixtures.

File: src/sas/qtgui/Calculators/media/sld_calculator_help.rst
Comment (krzywon) [inline src/sas/qtgui/Calculators/media/sld_calculator_help.rst:46]: If we are describing the inputs, would a description of the outputs also make sense?

File: src/sas/qtgui/Calculators/media/sld_calculator_help.rst
Comment (krzywon) [inline src/sas/qtgui/Calculators/media/sld_calculator_help.rst:49]: All instances of 'for example, let us consider' feel redundant and could probably be removed.

File: src/sas/qtgui/Calculators/media/sld_calculator_help.rst
Comment (krzywon) [inline src/sas/qtgui/Calculators/media/sld_calculator_help.rst:71]: This is an important statement that could easily be missed if it is not highlighted (bold/italic/etc.)

File: src/sas/qtgui/Calculators/media/sld_calculator_help.rst
Comment (krzywon) [inline src/sas/qtgui/Calculators/media/sld_calculator_help.rst:73]: Noting the `%` symbol must come before `wt` or `vol` might be important for people used to `wt%` and `vol%`.

**Review by krzywon — COMMENTED** [review]
This is a vast improvement over what already existed. I offered a few suggestions, but nothing that would hold this up.

File: src/sas/qtgui/Calculators/media/sld_calculator_help.rst
Comment (katieweigandt) [reply src/sas/qtgui/Calculators/media/sld_calculator_help.rst]: Updated

File: src/sas/qtgui/Calculators/media/sld_calculator_help.rst
Comment (katieweigandt) [reply src/sas/qtgui/Calculators/media/sld_calculator_help.rst]: Okay, okay... some definitions provided

File: src/sas/qtgui/Calculators/media/sld_calculator_help.rst
Comment (katieweigandt) [reply src/sas/qtgui/Calculators/media/sld_calculator_help.rst]: Agreed. Italicized.

**Comment by katieweigandt** [comment]
@krzywon I updated the the document with your suggestions. I'd appreciate it if you would take another look.

**Comment by katieweigandt** [comment]
I found two more formatting issues in the recent build. I will fix and update the pull request.

**Comment by wpotrzebowski** [comment]
Well done @katieweigandt! And pretty amazing that SLD calculator can do all of this:D

I haven't reviewed it properly yet but I noticed that it says that X-ray wavelength is required parameter and I think it is optional (or at least treated the same as Neutron wavelength).  Default starting value can also be provide. 

Volume fraction examples doesn't seem to work for me. When I enter 50%vol H2O // D2O I get an error:
`06:40:21 - ERROR: Traceback (most recent call last): File "periodictable/formulas.py", line 262, in formula File "periodictable/formulas.py", line 881, in parse_formula File "pyparsing/util.py", line 256, in _inner File "pyparsing/core.py", line 1187, in parse_string File "pyparsing/core.py", line 974, in _parseCache File "pyparsing/core.py", line 850, in _parseNoCache File "pyparsing/core.py", line 4037, in parseImpl File "pyparsing/core.py", line 974, in _parseCache File "pyparsing/core.py", line 850, in _parseNoCache File "pyparsing/core.py", line 5236, in parseImpl File "pyparsing/core.py", line 974, in _parseCache File "pyparsing/core.py", line 850, in _parseNoCache File "pyparsing/core.py", line 4293, in parseImpl File "pyparsing/core.py", line 974, in _parseCache File "pyparsing/core.py", line 889, in _parseNoCache File "pyparsing/core.py", line 293, in wrapper File "periodictable/formulas.py", line 788, in convert_by_volume File "periodictable/formulas.py", line 178, in _mix_by_volume_pairs ValueError: Need the mass density of H2O During handling of the above exception, another exception occurred: Traceback (most recent call last): File "sas/qtgui/Calculators/SldPanel.py", line 196, in recalculateSLD File "sas/qtgui/Calculators/SldPanel.py", line 57, in neutronSldAlgorithm File "periodictable/util.py", line 96, in _require_kwds File "periodictable/nsf.py", line 872, in neutron_scattering File "periodictable/formulas.py", line 271, in formula ValueError: Need the mass density of H2O`

Does Mass density need to be modified too? I don't see how this can be done as it only accepts single number 

Also (but not as a part of this PR) it would be great to turn all of this examples into unit tests.

**Comment by pkienzle** [comment]
Changes allowing wt% and vol% are in https://github.com/pkienzle/periodictable/pull/65. I'm leaving it up for a bit  before merging to see if there are any comments.

I also have an update to the mass and abundance tables pending. Unfortunately this will change the output from the formula calculations, which may break tests in dependent packages such as sasview and refl1d. I will probably do a point release with the wt%/vol% fix before doing a major release that changes all the values.

File: src/sas/qtgui/Calculators/media/sld_calculator_help.rst
Comment (krzywon) [reply src/sas/qtgui/Calculators/media/sld_calculator_help.rst:76]: @pkienzle submitted a [pull request](https://github.com/pkienzle/periodictable/pull/65) to allow both syntaxes. My original comment will be outdated by the time this becomes relevant. Calling this resolved.

**Review by krzywon — APPROVED** [review]

**Comment by butlerpd** [comment]
This should probably be against 6.0.0 as well.

QUESTION: @krzywon approved, but not clear how @wpotrzebowski and @pkienzle comments have been addressed or impact this PR. Is this ready to merge or is there something else that needs addressing?

**Comment by wpotrzebowski** [comment]
Volume fraction examples e.g `50%vol H2O // D2O` still don't work but they should once a new version of periodictable (including https://github.com/pkienzle/periodictable/pull/65) is released.

**Comment by katieweigandt** [comment]
I think I should comment out the part that is broken. Thanks for catching
that!  When we update the version of periodic table we use there are
several other features that exist and will need to be added to the
documentation.

**Comment by butlerpd** [comment]
vol% should be fixed for the release so that does not need to be removed but solvent fraction (for label hydrogen exchange) issue will not be done for this release.

**Review by butlerpd — CHANGES_REQUESTED** [review]
Several typos that should be corrected before merging (hence the "changes requested" but they are minor.
* Line 2 of `Description`. This may be an oxford comma type issue? But I don't think there should be a comma after including (including molecules, solutions/mixtures, etc). it is a comma separated list which when I was growing up did not start with a comma?
* Still in `Description`. What does `extensible` periodic table mean in this context. It is in the title of the reference but I assume that is more of a developer point than a user point? I think just omitting the word `extensible` would be less confusing to a general user? perhaps something like `which provides a periodic table populated with..`?
* Under `User Inputs` in the "Molecular Formula` section it might be nice to turn the reference to "Specifying Materials..." into a link to the section? 
* in the `Mass Density` section, the sentence should end `..of the material for which you are calculating the SLD.` the "of your material" is redundant.
* Under `Calculator Outputs` the sections on `Neutron SLD` and `X-ray SLD` "A required input for most scattering models [using in] neutron/x-ray scattering experiments." Probably should read `used in`? ... perhaps better would be `used in fitting data from....`
* Under `Specifying Materials or Mixtures in the Molecular Formula Field` in the `Multipart species` section "can be constructed [by as] a single" probably was meant to just be "by"? I would argue that "`using` a single" might be better?
* in the `isotopes` section second sentence, "isotopes of hydrogen which [c]an also be represented.." I believe the word is can not an.
* in the `Volume Fraction` section the parenthetical expression: "will are" could be "are" but I would suggest "will lead to" syntax errors.
* In the same section after the first example, I would add a comma after "100 mL with D2O[,] this can be.."
* Likewise after the second example, I would move the comma (and add "the") as in "For even more complicated solutions[,] parenthesis can be nested[,] and [the] number of atoms can be.." 
* In the `Solution Composition with Mass and Volume` I wonder if "volume" should be omitted? it is not true that you can mix the two as in 5g NaCl // 50ml H2O right? and none of the examples show volume being used.
* Same section, the subscripts did not work for H2O and D2O but did for NO3 so guessing there is an incorrect entry .. maybe a space after ``:sub:`2`?``
* A general note that the use of actual subscripts and superscripts vs not using them seems arbitrary?  even in that one formula?
* In the `Biomolecules` section there should be an and IMO after peptides. Whether or not to leave the coma I will bite my tongue...

**Review by butlerpd — APPROVED** [review]
except for the 
> an periodic table
which I can live with, this is now ready to merge

**Pull request merged**
